Mathlib/LinearAlgebra/TensorProduct.lean

```diff
@@ -91,11 +91,20 @@ section Module
 def addMonoid : AddMonoid (M ⊗[R] N) :=
   { (addConGen (TensorProduct.Eqv R M N)).addMonoid with }
 
+protected instance add : Add (M ⊗[R] N) :=
+  (addConGen (TensorProduct.Eqv R M N)).hasAdd
+
 instance addZeroClass : AddZeroClass (M ⊗[R] N) :=
-  { (addConGen (TensorProduct.Eqv R M N)).addMonoid with }
+  { (addConGen (TensorProduct.Eqv R M N)).addMonoid with
+    toAdd := TensorProduct.add _ _ }
```

Comment:
Could you (in a comment) explain why making the addition explicit is needed? I suppose it's because we want to have the actual ancestor of the class, so unification is faster.

Idem below.

Reply:
In fact, there should probably be a library note explaining the optimal way to do this kind of inheritance, since it changed quite a bit since Lean 3.